lib/orderbook/OrderBook.ts

```diff
@@ -431,14 +431,51 @@ class OrderBook extends EventEmitter {
   /**
    * Removes an order from the order book by its local id. Throws an error if the specified pairId
    * is not supported or if the order to cancel could not be found.
+   * @returns any quantity of the order that was on hold and could not be immediately removed.
    */
   public removeOwnOrderByLocalId = (localId: string) => {
-    const order = this.localIdMap.get(localId);
-    if (!order) {
+    const orderIdentifier = this.localIdMap.get(localId);
+    if (!orderIdentifier) {
       throw errors.LOCAL_ID_DOES_NOT_EXIST(localId);
     }
 
-    this.removeOwnOrder(order.id, order.pairId);
+    const order = this.getOwnOrder(orderIdentifier.id, orderIdentifier.pairId);
+    if (order.hold) {
+      let remainingHold = order.hold;
+      // we can't remove the entire order as some of it is on hold, start by removing any available portion
+      const availableQuantity = order.quantity - order.hold;
+      if (availableQuantity) {
+        this.removeOwnOrder(orderIdentifier.id, orderIdentifier.pairId, availableQuantity);
+      }
+
+      const cleanup = (quantity: number) => {
+        remainingHold -= quantity;
+        if (remainingHold === 0) {
```

Comment:
The swaps should have timeouts that cause them to fail after a certain time. My thinking is that if we ever get stuck here, it's a sign that something has gone seriously wrong elsewhere, and that timing out and removing the order despite a remaining hold may actually cover up the underlying issue.

Reply:
I see what you mean, I figure that we'll know if this happens because the remove order call will hang indefinitely. And yeah I think it'd make more sense to log and address this around the swaps.